comp/trace/agent/agent.go

```diff
@@ -98,6 +98,17 @@ func newAgent(deps dependencies) Component {
 		tagger:             deps.Tagger,
 		wg:                 sync.WaitGroup{},
 	}
+	statsdCl, err := setupMetrics(ag.statsd, ag.config, ag.telemetryCollector)
```

Comment:
Nit:
I don't think we need `statsd` in the `agent` struct anymore now that it's not being used anywhere but here.

Reply:
`pkgagent.NewAgent` still needs a statsd client. Do you mean using a noop client here?

Reply:
NVM, done in https://github.com/DataDog/datadog-agent/pull/26105/commits/6ee81d4a3fadf1f712c5edfc02df97b05572a10d